src/init.cpp

```diff
@@ -302,8 +304,8 @@ BOOL CtrlHandler( DWORD fdwCtrlType )
     case CTRL_LOGOFF_EVENT: 
     case CTRL_SHUTDOWN_EVENT: 
       fRequestShutdown = true;
-      std::cout << _("Stopping node.  This may take a while, be patient!") << std::endl;
-      while (!fShutdownCompleted)
+      std::cout << _("Shutting down node.  This may take a while, be patient!") << std::endl;
+      while (!fShutdownCompleted && (GetTime() - nStart < nMaxWait))
```

Comment:
Just one note that it might make sense to move the timeout check inside the loop so that you can log on timeout before letting the method continue, a la:
````
while (!fShutdownCompleted) {
    if (GetTime() - nStart < nMaxWait) {
        // log
        break;
    }
    MilliSleep(100);
}
````